tests/phpunit/elementor/modules/usage/collection/test-document-settings-usage.php

```diff
@@ -0,0 +1,64 @@
+<?php
+namespace Elementor\Tests\Phpunit\Elementor\Modules\Usage;
+
+use Elementor\Modules\Usage\Collection\DocumentSettingsUsage;
+use ElementorEditorTesting\Elementor_Test_Base;
+
+class TestDocumentSettingsUsage extends Elementor_Test_Base {
+	const DOCUMENTS_TEST_COUNT = 2;
+	const DOCUMENT_TEST_SETTINGS = [
+		'background_background' => 'red',
+		'hide_title' => 'yes',
+	];
+
+	private function create_document() {
+		$document = $this->factory()->create_post();
+		$document->save( [
+			'settings' => self::DOCUMENT_TEST_SETTINGS,
+		] );
+
+		return $document;
+	}
+
+	public function test_add() {
+		// Arrange.
+		$collection = new DocumentSettingsUsage( [] );
+
+		$document = $this->create_document();
+
+		for( $i = 0 ; $i < self::DOCUMENTS_TEST_COUNT ; ++$i ) {
+			$collection = $collection->add( $document );
+		}
+
+		// Ensure collection added.
+		$this->assertEquals( self::DOCUMENTS_TEST_COUNT, $collection->get( 'wp-post' )['background_background'] );
+	}
+
+	public function test_remove() {
+		// Arrange.
+		$collection = new DocumentSettingsUsage( [] );
+
+		$document = $this->create_document();
+
+		for( $i = 0 ; $i < self::DOCUMENTS_TEST_COUNT ; ++$i ) {
+			$collection = $collection->add( $document );
+		}
+
+		$differ_count = 0;
```

Comment:
Start with the full count and decrease on every iteration

Reply:
Handled: https://github.com/elementor/elementor/pull/13408/commits/72d34e808c04d2485e7cbff293ec534a9678b4df